src/compiler/checker.ts

```diff
@@ -10100,7 +10109,11 @@ namespace ts {
                     if (typeVariable !== mappedTypeVariable) {
                         return mapType(mappedTypeVariable, t => {
                             if (isMappableType(t)) {
-                                return instantiateAnonymousType(type, createReplacementMapper(typeVariable, t, mapper));
+                                const replacementMapper = createReplacementMapper(typeVariable, t, mapper);
+                                return isArrayType(t) ? createArrayType(instantiateMappedTypeTemplate(type, numberType, /*isOptional*/ true, replacementMapper)) :

```

Comment:
isOptional doesn’t have any meaning for arrays, does it? Why not pass false if that’s true?

Reply:
The `isOptional` indicates whether a `-?` modifier on the mapped type should strip `undefined` from the source type. We want to do that when the source type originates in an optional property, but we also want to do it when the source type is an array element type.